presto-main/src/main/java/com/facebook/presto/sql/planner/StatisticsAggregationPlanner.java

```diff
@@ -0,0 +1,251 @@
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class StatisticsAggregationPlanner
+{
+    private final SymbolAllocator symbolAllocator;
+    private final Metadata metadata;
+
+    public StatisticsAggregationPlanner(SymbolAllocator symbolAllocator, Metadata metadata)
+    {
+        this.symbolAllocator = requireNonNull(symbolAllocator, "symbolAllocator is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+    }
+
+    public CreateStatisticAggregationsResult createStatisticsAggregation(TableStatisticsMetadata statisticsMetadata, Map<String, Symbol> columnToSymbolMap)
+    {
+        StatisticAggregationsDescriptor.Builder<Symbol> descriptor = StatisticAggregationsDescriptor.builder();
+
+        List<String> groupingColumns = statisticsMetadata.getGroupingColumns();
+        List<Symbol> groupingSymbols = groupingColumns.stream()
+                .map(columnToSymbolMap::get)
+                .collect(toImmutableList());
+
+        for (int i = 0; i < groupingSymbols.size(); i++) {
+            descriptor.addGrouping(groupingSymbols.get(i), groupingColumns.get(i));
+        }
+
+        ImmutableMap.Builder<Symbol, AggregationNode.Aggregation> aggregations = ImmutableMap.builder();
+
+        for (TableStatisticType type : statisticsMetadata.getTableStatistics()) {
+            if (type != ROW_COUNT) {
+                throw new PrestoException(NOT_SUPPORTED, "Table-wide statistic type not supported: " + type);
+            }
+        }
+
+        FunctionRegistry functionRegistry = metadata.getFunctionRegistry();
+        if (!statisticsMetadata.getTableStatistics().isEmpty()) {
+            QualifiedName count = QualifiedName.of("count");
+            AggregationNode.Aggregation aggregation = new AggregationNode.Aggregation(
+                    new FunctionCall(count, ImmutableList.of()),
+                    functionRegistry.resolveFunction(count, ImmutableList.of()),
+                    Optional.empty());
+            Symbol symbol = symbolAllocator.newSymbol("rowCount", BIGINT);
+            aggregations.put(symbol, aggregation);
+            descriptor.addTableStatistic(symbol, ROW_COUNT);
+        }
+
+        ImmutableMap.Builder<Symbol, Expression> projections = ImmutableMap.builder();
+        for (ColumnStatisticMetadata columnStatisticMetadata : statisticsMetadata.getColumnStatistics()) {
+            String columnName = columnStatisticMetadata.getColumnName();
+            ColumnStatisticType statisticType = columnStatisticMetadata.getStatisticType();
+            Symbol inputSymbol = columnToSymbolMap.get(columnName);
+            verify(inputSymbol != null, "inputSymbol is null");
+            Type inputType = requireNonNull(symbolAllocator.getTypes().get(inputSymbol), "inputType is null");
```

Comment:
line above uses `verify`, this one uses `requireNonNull` for checking nullity, but i see no reason for it

Reply:
Replaced with a verify. Just in case the type for the symbol is not there.